src/jit/importer.cpp

```diff
@@ -16440,6 +16439,33 @@ bool Compiler::impReturnInstruction(BasicBlock* block, int prefixFlags, OPCODE&
 #endif
                 }
 
+                // If we are inlining a method that returns a struct byref, check whether we are "reinterpreting" the
+                // struct.
+                if ((returnType == TYP_BYREF) && (info.compRetType == TYP_BYREF) && (op2->OperGet() == GT_ADDR))
+                {
+                    GenTree* op2Child = op2->gtGetOp1();
```

Comment:
I don't like the name `op2Child` here,  Since we must have a `GT_ADDR` the name `addrChild` would be better

Reply:
Done.